Zend/tests/match/033.phpt

```diff
@@ -0,0 +1,34 @@
+--TEST--
+Test match block warning with unnecessary return value
+--FILE--
+<?php
+
+function test($value) {
+    match ($value) {
+        1 => {
+            echo '1';
+            echo '2';
```

Comment:
This segfaults if the code `break`s when the return value is used.
Here, the assignment to T4 is skipped over, so opcache segfaults trying to analyze this.

```
<?php
/*
0000 CV0($x) = RECV 1
0001 CV1($y) = RECV 2
0002 JMP 0003
0003 ECHO string("Start
")
0004 JMP 0007
0005 T4 = QM_ASSIGN string("intended return value")
0006 JMP 0007
0007 ASSIGN CV2($result) T4
0008 INIT_FCALL 1 96 string("var_dump")
0009 SEND_VAR CV2($result) 1
0010 DO_ICALL
0011 RETURN null
LIVE RANGES:
     4: 0006 - 0007 (tmp/var)
In function ::test (before dfa):
var op2 of 3 (ZEND_ASSIGN) does not use/def an ssa var
 */
function test ($x, $y) {
    $result = match(true) {
        default => {
            echo "Start\n";
            break;  // or if (true) { break 1; }
            'intended return value'
        }
    };
    var_dump($result);
}
```

Reply:
Oh, what would I do without you 😄 We should disallow `break` when we return a value then.

Reply:
So I guess we should also rethink when `goto` is allowed.

```php
$y = match ($x) {
    default => {
        // Is this valid?
        goto label;
    },
};

label:
```

```php
while (...) {
    $y = match ($x) {
        default => {
            // Is this valid?
            goto label;
        },
    };
}

label:
```

Reply:
`goto` and `break` are now disallowed in match expressions that a value.